packages/ui-drilldown/src/Drilldown/DrilldownOption/props.ts

```diff
@@ -200,7 +205,11 @@ const propTypes: PropValidators<PropKeys> = {
   descriptionRole: PropTypes.string,
   onOptionClick: PropTypes.func,
   defaultSelected: PropTypes.bool,
-  elementRef: PropTypes.func
+  elementRef: PropTypes.func,
+  shouldCloseOnClick: PropTypes.oneOfType([
```

Comment:
is there a reason this can't be simply just a boolean?

Reply:
IMO this should be an enum, it has 3 values. While its possible with a boolean, it could be easily a source of bugs when e.g. `undefined` is auto cast to boolean

Reply:
so you suggest something like `'auto' | 'always' | 'never'`?

Reply:
yes